tests/Escaper/EscaperTests.cpp

```diff
@@ -0,0 +1,223 @@
+    static constexpr uint8_t TEST_ARRAY_2[] = { STOP_BYTE, 21,  87,         START_BYTE, ESCAPE_BYTE, 43, 98,
+                                                122,       197, START_BYTE, 252,        2,           8 };
+
+    Escaper* escaper{ nullptr };
+
+    static constexpr size_t ENCODED_PACKET_MAX_SIZE{ 20 };
+    static constexpr size_t DECODED_PACKET_MAX_SIZE{ 20 };
+
+    uint8_t* encodedPacketBuffer;
+    uint8_t* decodedPacketBuffer;
+
+    void setup()
+    {
+        // always create "fresh" buffers and structure before test
+        encodedPacketBuffer = new uint8_t[ENCODED_PACKET_MAX_SIZE];
+        decodedPacketBuffer = new uint8_t[DECODED_PACKET_MAX_SIZE];
+        escaper = new Escaper{ .m_encoded_packet_buffer = encodedPacketBuffer,
+                               .m_encoded_packet_max_size = ENCODED_PACKET_MAX_SIZE,
+                               .m_decoded_packet_buffer = decodedPacketBuffer,
+                               .m_decoded_packet_max_size = DECODED_PACKET_MAX_SIZE };
+    }
+
+    void teardown()
+    {
+        delete escaper;
+        delete[] encodedPacketBuffer;
+        delete[] decodedPacketBuffer;
+    }
+
+    static void ReceivePacketFn(uint8_t* const buffer, size_t length) {}
+
+    void VerifyEncoding(const Escaper* const escaper, size_t encodedLength)
+    {
+        BYTES_EQUAL_TEXT(START_BYTE, escaper->m_encoded_packet_buffer[0], "No START_BYTE");
+        BYTES_EQUAL_TEXT(STOP_BYTE, escaper->m_encoded_packet_buffer[encodedLength - 1], "No STOP_BYTE");
+
+        for(uint8_t j = 1; j < encodedLength - 1; j++) {
+            switch(escaper->m_encoded_packet_buffer[j]) {
+                case START_BYTE:
+                    FAIL("!!!! Non-escaped START_BYTE !!!! \n\r");
+                    break;
+                case STOP_BYTE:
+                    FAIL("!!!! Non-escaped START_BYTE !!!! \n\r");
+                    break;
+                case ESCAPE_BYTE:
+                    switch(escaper->m_encoded_packet_buffer[j + 1]) {
+                        case START_BYTE:
+                        case STOP_BYTE:
+                        case ESCAPE_BYTE:
+                            j++;
+                            break;
+                        default:
+                            FAIL("!!!! Non-escaped ESCAPE_BYTE !!!!");
+                            break;
+                    }
+            }
+        }
+    }
+
+    size_t SetupParsing(Escaper* const escaper, const uint8_t* const buffer, int size)
+    {
+        size_t index{ 0 }, encodedLength;
+        Escaper_start_encoder(escaper);
+        encodedLength = Escaper_encode_packet(escaper, buffer, size, &index);
+        VerifyEncoding(escaper, encodedLength);
+
+        return encodedLength;
+    }
+
+    void VerifyParsing(const Escaper* const escaper, const uint8_t* const buffer, int size)
+    {
+        for(int j = 0; (j < size && j < sizeof(escaper->m_decoded_packet_buffer)); j++) {
```

Comment:
sizeof cannot be used here (`sizeof(pointer_typer`) is equal to 8 on linux x86_64)